bdk_core/src/coin_select.rs

```diff
@@ -173,13 +233,21 @@ impl CoinSelector {
         }
 
         let weight_with_drain = base_weight + self.opts.drain_weight;
+
         let target_fee_with_drain = ((self.opts.target_feerate * weight_with_drain as f32).ceil()
             as u64)
             .max(self.opts.min_absolute_fee);
         let target_fee_without_drain = ((self.opts.target_feerate * base_weight as f32).ceil()
             as u64)
             .max(self.opts.min_absolute_fee);
 
+        let long_term_fee_with_drain = ((self.opts.long_term_feerate * weight_with_drain as f32)
+            .ceil() as u64)
+            .max(self.opts.min_absolute_fee);
+        let long_term_free_without_drain = ((self.opts.long_term_feerate * base_weight as f32)
```

Comment:
```suggestion
        let long_term_fee_without_drain = ((self.opts.long_term_feerate * base_weight as f32)
```

Reply:
This doesn't make sense to me. Why are we using the `long_term_feerate` against the transaction's weight. The transactions is being broadcast now, not in the "long term". Isn't `long_term_fee_without_drain == target_fee_without_drain`.

Reply:
You are right, only the weight of the inputs should be considered.

Reply:
why would you consider the weight of the inputs when their fee is also not being paid in the "long term", they are being paid now. Do you mean outputs?

Reply:
`waste` is the wasted cost of not spending inputs now, so we want to compare the "cost" of spending inputs now vs later.

Reply:
ahhh understood thanks